packages/cli/src/utils/generate.ts

```diff
@@ -124,29 +124,48 @@ async function copyTheme() {
 }
 
 function mergeCMSFile(fileName: string) {
+  const customFilePath = `${userCMSDir}/${fileName}`
+  const coreFilePath = `${coreCMSDir}/${fileName}`
+
+  const coreFile = readFileSync(coreFilePath, 'utf8')
+  const coreJSON = JSON.parse(coreFile)
+
+  let output = []
+
   // TODO: create a validation when has the cms files but doesn't have a component for then
-  if (existsSync(userCMSDir) && readdirSync(userCMSDir).length > 0) {
-    const coreContentTypes = readFileSync(`${coreCMSDir}/${fileName}`, 'utf8')
-    const customContentTypes = readFileSync(`${userCMSDir}/${fileName}`, 'utf8')
-    const coreContentTypesJSON = JSON.parse(coreContentTypes)
-    const customContentTypesJSON = JSON.parse(customContentTypes)
-
-    const mergeContentTypes = [
-      ...coreContentTypesJSON,
-      ...customContentTypesJSON,
-    ]
+  if (existsSync(customFilePath)) {
+    const customFile = readFileSync(customFilePath, 'utf8');
+    let customJSON;
 
     try {
-      writeFileSync(
-        `${tmpCMSDir}/${fileName}`,
-        JSON.stringify(mergeContentTypes)
-      )
-      console.log(
-        `${chalk.green('success')} - CMS file ${chalk.dim(fileName)} created`
+      customJSON = JSON.parse(customFile)
+    } catch(SyntaxError) {
+      console.info(
+        `${chalk.red(
+          'error'
+        )} - ${fileName} is a malformed JSON file, ignoring its contents.`
       )
-    } catch (err) {
-      console.error(`${chalk.red('error')} - ${err}`)
+      customJSON = []
     }
+
+    output = [
+      ...coreJSON,
+      ...customJSON,
+    ]
```

Comment:
I think we need to double-check the intention behind this logic. 

This is adding a new `content-types` object in the array. Do we want it? or do we need a deep merge here for the sake of overriding core CMS `content-types` with the custom ones?
<img width="313" alt="Screenshot 2023-04-05 at 17 35 31" src="https://user-images.githubusercontent.com/11325562/230205389-ff2dd2eb-ca69-4a74-8b51-fbbf956ef47a.png">

Reply:
Yes, I didn't plan to change this logic, and I think that what we want is the first option indeed.